lib/spack/spack/cmd/help.py

```diff
@@ -22,16 +22,27 @@
 # License along with this program; if not, write to the Free Software
 # Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
 ##############################################################################
+import sys
+
 description = "get help on spack and its commands"
+section = "help"
+level = "short"
 
 
 def setup_parser(subparser):
-    subparser.add_argument('help_command', nargs='?', default=None,
-                           help='command to get help on')
+    help_cmd_group = subparser.add_mutually_exclusive_group()
+    help_cmd_group.add_argument('help_command', nargs='?', default=None,
+                                help='command to get help on')
```

Comment:
Good catch.  There used to be `spack help --spec`, which was mutually exclusive from the others.   It's gone now.